internal/ast/utilities.go

```diff
@@ -80,6 +80,17 @@ func PositionIsSynthesized(pos int) bool {
 	return pos < 0
 }
 
+func FindLastVisibleNode(nodes []*Node) *Node {
+	fromEnd := 1
+	for fromEnd <= len(nodes) && nodes[len(nodes)-fromEnd].Flags&NodeFlagsSynthesized != 0 {
```

Comment:
I wonder if `Synthesized` is the wrong flag for this and we need something else like `Virtual` instead. We seem to be running out of space in `NodeFlags`, but it might be worth seeing if there are any flags we can reclaim or move. For example, `NodeFlagsNamespace`, `NodeFlagsNestedNamespace`, and `NodeFlagsGlobalAugmentation` only ever apply to a `ModuleDeclaration`, so it might make more sense to move those to a separate enum and field on `ModuleDeclaration` instead. We may not even need `NodeFlagsGlobalAugmentation` as that can be trivially inferred from the name of the module declaration (`global`).

Reply:
A number of these flags were previously computed by the checker so emit could use them. I think a lot of them can be eliminated now.

Reply:
We've discussed deprecating `module id {}` in favor of `namespace id {}` for a while, especially in light of the [Module Declarations Proposal](https://github.com/tc39/proposal-module-declarations). Once we've finished the port, we should give some thought to splitting `namespace id {}` out to `NamespaceDeclaration` and only supporting `module "string" {}` and `global {}` for `ModuleDeclaration`.